benchmarks/smallmatvec/manual_versions/libxsmm/README.md

```diff
@@ -0,0 +1,52 @@
+
+This directory allows the performance of the default matrix vector
+kernel benchmark to be compared with a libxsmm version and the kinner
+version with the matrix data reordered. In the libxsmm version the
+matmul intrinsic is replaced and an explicit just-in-time compilation
+call is placed outside of the k-loop.
+
+The libxsmm version was originally proposed, implemented and tested
+(on the full LFRic code) by Wolfgang Hayek where he found a
+significant performance improvement. This version differs slightly
+from his version as he also included a special case when the number of
+dofs in both cases is 1. In this benchmark we do not have this
+situation so can safely ignore this additional
+optimisation. Optimising for particular numbers of dofs is being done
+but is being treated as a separate optimisation.
+
+The files in this directory are
+
+matrix_vector_kernel_mod.f90: the original matrix vector kernel with
+metadata removed to reduce the number of modules needed to compile the
```

Comment:
Hanging indent to match those below please.

Reply:
oops